src/core/lib/event_engine/ares_resolver.cc

```diff
@@ -599,9 +603,9 @@ void AresResolver::OnWritable(FdNode* fd_node, absl::Status status) {
   GRPC_TRACE_LOG(cares_resolver, INFO)
       << "(EventEngine c-ares resolver) OnWritable: fd: " << fd_node->as
       << "; request:" << this << "; status: " << status;
-  if (status.ok() && !shutting_down_) {
+  if (status.ok() && !shutting_down_ && channel_ != nullptr) {
     ares_process_fd(channel_, ARES_SOCKET_BAD, fd_node->as);
-  } else {
+  } else if (fd_node->polled_fd->IsCurrent() && channel_ != nullptr) {
```

Comment:
I think this leaves paths through the resolver where channels can become null, and outstanding requests are not canceled. Maybe in the pre-fork logic, it could cancel any outstanding requests before destroying the channel?

Reply:
Right! Thanks for pointing this out.

I rewrote the code to explicitly report status provided to callbacks. Now EE will use Cancelled (for compliance with Python expectations) and specific message.

Note that the test verifies that callback was notified.